src/commands/add_command/stake_proposal/transfer_near_tokens_type/mod.rs

```diff
@@ -180,28 +155,60 @@ impl TransferNEARTokensAction {
 
 impl TransferNEARTokensAction {
     fn input_amount(
-        account_balance: Option<crate::common::NearBalance>,
-    ) -> crate::common::NearBalance {
-        match account_balance {
-            Some(account_balance) => loop {
-                let input_amount: crate::common::NearBalance = Input::new()
-                            .with_prompt("How many NEAR Tokens do you want to transfer? (example: 10NEAR or 0.5near or 10000yoctonear)")
-                            .with_initial_text(format!("{}", account_balance))
-                            .interact_text()
+        connection_config: Option<crate::common::ConnectionConfig>,
+        sender_account_id: near_primitives::types::AccountId,
+    ) -> color_eyre::eyre::Result<crate::common::TransferAmount> {
+        match connection_config {
+            Some(connection_config) => {
+                let account_transfer_allowance = crate::common::get_account_transfer_allowance(
+                    &connection_config,
+                    sender_account_id,
+                )?;
+                println! {"{}", &account_transfer_allowance};
+                loop {
+                    let input_amount: crate::common::NearBalance = Input::new()
+                        .with_prompt("How many NEAR Tokens do you want to stake? (example: 10NEAR or 0.5near or 10000yoctonear)")
+                        .interact_text()
+                        .unwrap();
+                    if let Ok(transfer_amount) = crate::common::TransferAmount::from(
+                        input_amount.clone(),
+                        &account_transfer_allowance,
+                    ) {
+                        break Ok(transfer_amount);
+                    } else {
+                        println!(
+                            "\nWARNING! There is only {} available for stake.",
+                            account_transfer_allowance.transfer_allowance()
```

Comment:
there should be a different formula there for staking allowance (`account_liquid_amount + account_locked_amount - transaction_fee`)